lib/index.js

```diff
@@ -114,18 +103,58 @@ function setupRateLimitEventExt(server, options) {
   };
 
   server.ext(event, (request, reply) => {
+    const options = parseRouteOptions(request);
+    if (!options.enabled) {
+      return reply.continue();
+    }
     // This handler is going to be called one time per registration of patova
-    getType(request, reply, (err, type) => {
-      extractKeyAndTakeToken(options.limitd, request, reply, type);
+    getType(options, request, reply, (err, type) => {
+      extractKeyAndTakeToken(options, request, reply, type);
     });
   });
+
+  server.ext('onPreResponse', (request, reply) => {
+    const requestLimit = request.plugins.patova && request.plugins.patova.limit;
+
+    if (requestLimit && requestLimit.conformant) {
+      const options = parseRouteOptions(request);
+      if (options.addHeaders) {
+        const headers = new RateLimitHeaders(
+            requestLimit.limit,
+            requestLimit.remaining,
+            requestLimit.reset);
+
+        Object.keys(headers).forEach(
+          key => setResponseHeader(request, key, headers[key]));
+      }
+    }
+    reply.continue();
+  });
+
+  function parseRouteOptions(request) {
+    const routeOptions = request.route.settings.plugins.patova;
+    if (routeOptions === undefined) {
+      return pluginOptions;
+    }
+    if (routeOptions._merged) { //already merged with previous pluginOptions;
+      return routeOptions;
+    }
+    request.route.settings.plugins.patova = Object.assign({_merged: true}, pluginOptions, {enabled: true}, routeOptions);
+    return request.route.settings.plugins.patova;
+  }
+
 }
 
-exports.register = function (server, options, next) {
-  Joi.validate(options, schema, { abortEarly: false }, (err, processedOptions) => {
+const defaults = {
```

Comment:
does it make sense to use Joi `default`: https://github.com/hapijs/joi/blob/master/API.md#anydefaultvalue-description?